lib/rubocop/cop/rails/has_many_or_has_one_dependent.rb

```diff
@@ -39,17 +39,33 @@ class HasManyOrHasOneDependent < Cop
           (pair (sym :through) !nil)
         PATTERN
 
+        def_node_matcher :with_options_block?, <<-PATTERN
```

Comment:
I would ✂️ the `?` here, since the matcher captures a node, and doesn't return `true`/`false`. 🙂